### ISSUE TITLE
Backport PRs from main branch

### ISSUE THREAD

**Issue by stiiifff** [paritytech/trappist#175]
Backport latest merged PRs from `main` branch into `xcmv3-dev`.

- [x] https://github.com/paritytech/trappist/pull/166
- [x] https://github.com/paritytech/trappist/pull/165
- [x] https://github.com/paritytech/trappist/pull/163
- [x] https://github.com/paritytech/trappist/pull/173
- [x] https://github.com/paritytech/trappist/pull/174
- [x] https://github.com/paritytech/trappist/pull/176
- [x] https://github.com/paritytech/trappist/pull/185
- [x] https://github.com/paritytech/trappist/pull/190
- [x] https://github.com/paritytech/trappist/pull/206
- [x] https://github.com/paritytech/trappist/pull/207
- [x] https://github.com/paritytech/trappist/pull/211
- [x] https://github.com/paritytech/trappist/pull/220
- [x] https://github.com/paritytech/trappist/pull/223
- [x] https://github.com/paritytech/trappist/pull/224
- [x] https://github.com/paritytech/trappist/pull/226
- [x] https://github.com/paritytech/trappist/pull/230
- [x] https://github.com/paritytech/trappist/pull/221
- [x] Bump docker versions (PRs 218/219/232/233)

Hopefully, the commits can just be cherry-picked and re-applied on `xcmv3-dev`.

Depends on #171

**Comment by valentinfernandez1** [comment]
Should be relatively simple. The only things that might conflict with the the branch are the benchmarks of `drop_assets` and `asset_registry` since the weights are still using V1 format. But should be a simple fix.

So this should be merged after #171.

**Comment by valentinfernandez1** [comment]
This feature is done but #171 must be merged first.

**Comment by valentinfernandez1** [comment]
I will include the zombienet refactor #190 alongside the CI refactoring #174 in a single PR once the changes are merged into main.

Also some CI testing needs to be done since currently the ubuntu machine that executes the commands runs out of memory when running the commands.  We discussed with alex that some tweaking needs to be done in this section to fix that issue
```yaml
 - name: Free Disk Space (Ubuntu)
        uses: jlumbroso/free-disk-space@main
        with:
          android: true
          dotnet: false
          haskell: false
          large-packages: false
          swap-storage: false
```

**Comment by valentinfernandez1** [comment]
To anyone that needs to backport a feature in the future I will unify the format to be similar to the one used in cumulus

![Image](https://github.com/paritytech/trappist/assets/33705477/b9645559-9576-4cee-8526-123bdf1498f5)

This will make it easier to differentiate between new features and bakcports
